langchain/retrievers/azure_cognitive_search.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from typing import List, Dict, Optional
+
+import json
+import aiohttp
+import requests
+from pydantic import BaseModel, Extra, root_validator
+
+from langchain.schema import BaseRetriever, Document
+from langchain.utils import get_from_dict_or_env
+
+
+class AzureCognitiveSearchRetriever(BaseRetriever, BaseModel):
+    """Wrapper around Azure Cognitive Search."""
+    service_name: str  # name of Azure Cognitive Search service
+    index_name: str
+    azure_cognitive_search_api_key: Optional[str] = None
+    aiosession: Optional[aiohttp.ClientSession] = None
+
+    class Config:
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+    @root_validator(pre=True)
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key exists in environment."""
+        api_key = get_from_dict_or_env(
+            values, "azure_cognitive_search_api_key", "AZURE_COGNITIVE_SEARCH_API_KEY"
+        )
+        values["azure_cognitive_search_api_key"] = api_key
+        return values
+
+
+    def _build_search_url(self, api_version: str = "2020-06-30") -> str:
```

Comment:
should api_version be an instance variable? not sure how else someone would configure it

Reply:
I agree - done ✅